src/Gyro.cpp

```diff
@@ -34,67 +42,88 @@ void Gyro::startup()
     Wire.write(MPU6050_REG_WHO_AM_I);
     Wire.endTransmission(true);
     Wire.requestFrom(MPU6050_I2C_ADDR, 1, 1);
-    byte id = (Wire.read() >> 1) & 0x3F;    
+    byte id = Wire.read();
+    id = (id >> 1) & 0x3F;
     isPresent = (id == 0x34);
-    if (!isPresent) {
-        LOGV1(DEBUG_INFO, F("GYRO:: Not found!"));
+    if (!isPresent)
+    {
+        LOGV2(DEBUG_INFO, F("GYRO:: Not found! Expected 0x34 but got %x"), id);
         return;
     }
 
-    // Execute 1 byte write to MPU6050_REG_PWR_MGMT_1
+    // Execute 1 byte write to MPU6050_REG_PWR_MGMT_1 to disable sleep and set 8MHz clock
     Wire.beginTransmission(MPU6050_I2C_ADDR);
     Wire.write(MPU6050_REG_PWR_MGMT_1);
-    Wire.write(0);      // Disable sleep, 8 MHz clock
+    Wire.write(0); // Disable sleep, 8 MHz clock
     Wire.endTransmission(true);
 
-    // Execute 1 byte write to MPU6050_REG_PWR_MGMT_1
+    // Execute 1 byte read/write to MPU6050_REG_ACCEL_CONFIG to set 4g sensititvity
+    Wire.beginTransmission(MPU6050_I2C_ADDR);
+    Wire.write(MPU6050_REG_ACCEL_CONFIG);
+    Wire.endTransmission();
+    Wire.requestFrom(MPU6050_I2C_ADDR, 1);
+    byte x = Wire.read(); //the value of Register-28 is in x
+    x = (x & 0b11100111);     //clear values of Bit4 and Bit3 fo 2g sensitivity
```

Comment:
You say 4g sensitivity above, which one is it?

Reply:
Fixed